enhancements/helm3/console.md

```diff
@@ -0,0 +1,144 @@
+
+   * The public GitHub repository could be cloned into inside-the-network GitHub or Gitlab instance and configured to serve static content ( "Pages" ).
+   * The content of the chart repository could be crawled and served using a (containerized or external) HTTP server, e.g. nginx
+
+2. The URL serving the above static content would need to be passed to chart repository proxy running inside the cluster. 
+
+### Configuring Helm Chart Repository location
+
+Configuring Helm repository location could be modeled similar to [`OperatorSource`](https://github.com/operator-framework/operator-marketplace/blob/7d230952a1045624b7601b4d6e1d45b3def4cf76/deploy/crds/operators_v1_operatorsource_crd.yaml). 
+Due to future federated usecases ([ODC-2994](https://issues.redhat.com/browse/ODC-2994))), a cluster admin should be able to declare multiple chart repositories.
+
+A cluster-scoped, top-level `HelmRepository` resource)[https://github.com/openshift/api/tree/master/config/v1] should be added:
+
+```yaml
+apiVersion: config.openshift.io/v1
+kind: HelmRepository
+metadata:
+  name: cluster
+spec:
+  url: http://my.chart-repo.org/stable
+
+  # optional and only needed for UI purposes
+  displayName: myChartRepo
+
+  # optional and only needed for UI purposes
+  description: my private chart repo
+
+  # set to true if need to be disabled temporarly
+  disabled: true
+```
+
+The console operator would watch for changes on it and reconfigure the chart repository proxy. The console backend already implements a few helm endpoints (including the chart proxy), but the future might require to extract them into a separate service (e.g. to make openshift more modular). Hence, it would be good to define the configuration as the top-level resource detaching API from the implementation. Cluster admin would easily discover the new resource and manage it either via CLI or through UI. 
+
+![Helm Cluster Configuration](assets/openshift-administration-cluster-settings.png)
+
+#### Alternatives
+
+#### 1. The configuration could be embedded into cluster-wide [`Console` config](https://github.com/openshift/api/blob/master/config/v1/types_console.go#L26)
+
+Admins wouldn't be able to intuitively discover the operator config as a way to configure the Helm repository URLs. It becomes closely coupled with the console. Extracting Helm endpoints into a separate service would require moving the config as well.
+
+#### 2. The configuration could be embedded into [`Console` operator config](https://github.com/openshift/api/blob/master/operator/v1/types_console.go#L26)
+
+Conceptually, the Helm repository URL isn't really an operator configuration, hence this doesn't feel like the right place.
+This approach would have similar issues with the previous alternative - admins wouldn't be able to intuitively discover the operator config as a way to configure the Helm repository URLs.
+
+#### 3. OLM operator for Helm Configuration. 
```

Comment:
i view this as trying to determine who installs the helm related crd.

i feel like there is an option that wasn't discussed, but this is very similar to templates.

we have template.openshift.io, and we have a set of templates installed ootb.

can you add some language contrasting your use case with existing patterns around templates?

Reply:
I was not aware of that option.

We could add Helm chart repo config to a dedicated `helm.openshift.io` group and watch on changes there, but are we able to to manage the instance within UI? I could not find that option.

Reply:
> but are we able to to manage the instance within UI?

We can make an explicit change in the UI code to do that.